internal/layer2/announcer.go

```diff
@@ -20,6 +20,10 @@ type Announce struct {
 	ndps     map[int]*ndpResponder
 	ips      map[string]net.IP // svcName -> IP
 	ipRefcnt map[string]int    // ip.String() -> number of uses
+
+	// channel can block so do not produce to spamCh while holding the Mutex
+	// else you risk causing a dead lock
```

Comment:
```suggestion
	// This channel can block - do not write to it while holding the mutex
	// to avoid deadlocking.
```

- We should format the comments according to the existing style.
- Improve phrasing.